weave/storage.py

```diff
@@ -38,6 +38,8 @@ def split_path_dotfile(path, dotfile_name):
 
 
 def _get_name(wb_type: types.Type, obj: typing.Any) -> str:
+    if getattr(obj, "name", None) != None:
```

Comment:
This is the worst part of Weave.

We have to use != None, because None may be a BoxedNone, and there's no way to make `is None` work for that case.

I spent some time yesterday trying to figure out if I could get rid of the boxing so I wouldn't have to do this, but it caused issues in other Weave tests. And without boxing, we don't have ref tracking (so we'd never be able to have a ref that points to a None).

There's got to a better way that we'll find at some point, but I haven't found it!